dandiapi/api/views/users.py

```diff
@@ -22,14 +23,10 @@
 
 
 def _get_user_status(user: User):
-    # Workaround for https://github.com/dandi/dandi-archive/issues/1085
-    # TODO: remove/robustify some other way whenever underlying reason is
-    # identified
-    metadata = getattr(user, 'metadata', None)
-    if metadata:
-        return metadata.status
-    logger.error("User %s lacks .metadata. Returning user's status as INCOMPLETE", user)
-    return UserMetadata.Status.INCOMPLETE
+    try:
+        return user.metadata.status
+    except ObjectDoesNotExist:
```

Comment:
I think you can also do this, which makes it clear that the `metadata` field is what might not exist
```suggestion
    except UserMetadata.DoesNotExist:
```